### PR TITLE
zlib: add config.yml + fix hooks/delete fPIC if shared/fix cmake names in 1.2.8

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **zlib/1.2.8**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — APPROVED** [review]

File: recipes/zlib/1.2.8/conanfile.py [hunk 1]
```diff
@@ -108,9 +108,11 @@ def package(self):
                 self.copy(pattern="*.a", dst="lib", src=self._source_subfolder, keep_path=False)
 
     def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "ZLIB"
```
Comment (madebr) [inline recipes/zlib/1.2.8/conanfile.py:111]: Can you also explicitly set the `pkg_config` name?

Comment (SpaceIm) [reply recipes/zlib/1.2.8/conanfile.py:111]: What is the official pkg-config file name of zlib?

Comment (madebr) [reply recipes/zlib/1.2.8/conanfile.py:111]: Just zlib.
See the contents of this rpm
https://centos.pkgs.org/7/centos-x86_64/zlib-devel-1.2.7-18.el7.x86_64.rpm.html

Comment (SpaceIm) [reply recipes/zlib/1.2.8/conanfile.py:111]: Correct, in zlib's Makefile.in file:
```make
install-libs: $(LIBS)
	-@if [ ! -d $(DESTDIR)$(exec_prefix)  ]; then mkdir -p $(DESTDIR)$(exec_prefix); fi
	-@if [ ! -d $(DESTDIR)$(libdir)       ]; then mkdir -p $(DESTDIR)$(libdir); fi
	-@if [ ! -d $(DESTDIR)$(sharedlibdir) ]; then mkdir -p $(DESTDIR)$(sharedlibdir); fi
	-@if [ ! -d $(DESTDIR)$(man3dir)      ]; then mkdir -p $(DESTDIR)$(man3dir); fi
	-@if [ ! -d $(DESTDIR)$(pkgconfigdir) ]; then mkdir -p $(DESTDIR)$(pkgconfigdir); fi
	cp $(STATICLIB) $(DESTDIR)$(libdir)
	chmod 644 $(DESTDIR)$(libdir)/$(STATICLIB)
	-@($(RANLIB) $(DESTDIR)$(libdir)/libz.a || true) >/dev/null 2>&1
	-@if test -n "$(SHAREDLIBV)"; then \
	  cp $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir); \
	  echo "cp $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir)"; \
	  chmod 755 $(DESTDIR)$(sharedlibdir)/$(SHAREDLIBV); \
	  echo "chmod 755 $(DESTDIR)$(sharedlibdir)/$(SHAREDLIBV)"; \
	  rm -f $(DESTDIR)$(sharedlibdir)/$(SHAREDLIB) $(DESTDIR)$(sharedlibdir)/$(SHAREDLIBM); \
	  ln -s $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir)/$(SHAREDLIB); \
	  ln -s $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir)/$(SHAREDLIBM); \
	  ($(LDCONFIG) || true)  >/dev/null 2>&1; \
	fi
	cp zlib.3 $(DESTDIR)$(man3dir)
	chmod 644 $(DESTDIR)$(man3dir)/zlib.3
	cp zlib.pc $(DESTDIR)$(pkgconfigdir)
	chmod 644 $(DESTDIR)$(pkgconfigdir)/zlib.pc
```

Ok I can enforce zlib name for pkgconfig.

File: recipes/zlib/1.2.8/conanfile.py [hunk 2]
```diff
@@ -2,12 +2,10 @@
 import stat
 import shutil
 from conans import ConanFile, tools, CMake, AutoToolsBuildEnvironment
-from conans.errors import ConanException, NotFoundException
 
 
 class ZlibConan(ConanFile):
     name = "zlib"
-    version = "1.2.8"
```
Comment (uilianries) [inline recipes/zlib/1.2.8/conanfile.py:10]: You can keep version here, as this recipe is specific for a single version.

**Review by SSE4 — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**